lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -386,7 +447,7 @@ def _requires_modification(self):
             'EngineVersion': self.cache_engine_version
         }
         for key, value in modifiable_data.items():
-            if value is not None and self.data[key] != value:
+            if value and self.data[key] != value:
```

Comment:
the change from `value is not None` is very dangerous if any value can be `False` instead.